acapy_cache_redis/v0_1/redis_base_cache.py

```diff
@@ -65,13 +69,47 @@ def __init__(self, root_profile: Profile) -> None:
         # Setup the aioredis instance
         self.pool = aioredis.ConnectionPool.from_url(
             self.connection,
-            max_connections=max_connections,
-            username=username,
-            password=password,
-            # ssl_ca_certs=ca_cert,
+            max_connections=self.max_connections,
+            username=self.username,
+            password=self.password,
+            # ssl_ca_certs=self.ca_cert,
         )
         self.redis = aioredis.Redis(connection_pool=self.pool)
 
+    async def check_for_redis_cluster(self):
+        """
+        Check if connection corresponds to a cluster node. If so,
+        reassign redis to redis.asyncio.RedisCluster client.
+
+        """
+        try:
+            # Execute a redis SET command on a fake test_key prefix with b""
+            # value. In case, connection string is that of a single redis
+            # host then it will return None as it doesn't exists. Otherwise,
+            # it will raise a MOVED error.
+            fake_test_key = f"test_key_{str(uuid4())}"
+            await self.redis.set(fake_test_key, b"")
```

Comment:
Would it be better to set a quick expiration here? Something like a TTL of 1?

Reply:
Added expiry and fixed the int tests [awaiting approval to run]